packages/experimental/artifact/src/types/util.ts

```diff
@@ -9,20 +9,29 @@ import { toJsonSchema, type JsonSchemaType } from '@dxos/echo-schema';
 import { type Tool, type ToolExecutionContext, type ToolResult } from './tools';
 
 export type DefineToolParams<Params extends S.Schema.AnyNoContext> = {
+  /**
+   * The name of the tool (may include hyphens but not underscores).
+   */
   name: string;
+  caption?: string;
   description: string;
   schema: Params;
   execute: (params: S.Schema.Type<Params>, context: ToolExecutionContext) => Promise<ToolResult>;
 };
 
-export const defineTool = <Params extends S.Schema.AnyNoContext>({
-  name,
-  description,
-  schema,
-  execute,
-}: DefineToolParams<Params>): Tool => {
+export const parseToolName = (name: string) => {
+  return name.split('_').pop();
+};
+
+export const defineTool = <Params extends S.Schema.AnyNoContext>(
+  namespace: string,
```

Comment:
intuitively I thought tool ID would be the first argument - that's what I though when looking at examples first, and then at the definition. I would move the namespace into the params, so there's no ambiguity